### PR TITLE
Use GeneratedDllImport in System.Net.NetworkInformation

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
|     | Original (B) | Converted (B) | + (B) | + (%) |
| --- | -----------: | ------------: | ----: | ----: |
| windows | 69632 | 71680 | 2048 | 2.94 |
| windows R2R | 137728 | 147456 | 9728  | 7.06 |

/cc @elinor-fung @jkoritzinsky

### CODE REVIEW

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs [hunk 1]
```diff
@@ -52,8 +52,8 @@ public unsafe struct NetworkInterfaceInfo
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_EnumerateGatewayAddressesForInterface")]
         public static extern unsafe int EnumerateGatewayAddressesForInterface(void* context, uint interfaceIndex, delegate* unmanaged<void*, IpAddressInfo*, void> onGatewayFound);
 
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetNetworkInterfaces")]
-        public static unsafe extern int GetNetworkInterfaces(ref int count, ref NetworkInterfaceInfo* addrs, ref int addressCount, ref IpAddressInfo *aa);
+        [GeneratedDllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetNetworkInterfaces")]
+        public static unsafe partial int GetNetworkInterfaces(ref int count, ref NetworkInterfaceInfo* addrs, ref int addressCount, ref IpAddressInfo *aa);
```
Comment (elinor-fung) [inline src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: Can't we make this one blittable?

File: src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemIPGlobalStatistics.cs [hunk 2]
```diff
@@ -12,17 +12,19 @@ internal sealed class SystemIPGlobalStatistics : IPGlobalStatistics
 
         private SystemIPGlobalStatistics() { }
 
-        internal SystemIPGlobalStatistics(AddressFamily family)
+        internal unsafe SystemIPGlobalStatistics(AddressFamily family)
         {
-            uint result = Interop.IpHlpApi.GetIpStatisticsEx(out _stats, family);
+            uint result;
+            fixed (Interop.IpHlpApi.MibIpStats* pStats = &_stats)
```
Comment (elinor-fung) [inline src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemIPGlobalStatistics.cs:18]: I think libraries usually does the braces for fixed (even single line).

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemIPGlobalStatistics.cs:17]: Done.

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: Sadly, no. The `NetworkInterfaceInfo` has an embedded fixed array.

**Review by elinor-fung — APPROVED** [review]

File: src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs (hunk 1 above)
Comment (jkoritzinsky) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: Embedded `fixed` arrays of `byte`s shouldn't stop a type from being blittable.

Comment (jkoritzinsky) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: Managed array fields with `[MarshalAs(UnmanagedType.ByValArray)]` aren't blittable, but `fixed byte fieldName[n];` fields are.

Comment (AaronRobinsonMSFT) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: Sorry. In this case the `[MarshalAs(UnmanagedType.ByValArray)]` is what is being used.

Comment (jkoritzinsky) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: Where in the code is it using that? I just checked and it was using `fixed` for the three arrays.

Also, the runtime wouldn't let you have a pointer type to a struct that has array fields.

Can you point me to the code that uses the ByValArray annotation?

Comment (AaronRobinsonMSFT) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: https://github.com/dotnet/runtime/blob/2e2d1fb06693b1c44e1fb0ea48ad980cc1cc4500/src/libraries/Common/src/Interop/Windows/IpHlpApi/Interop.NetworkInformation.cs#L238-L292

Comment (AaronRobinsonMSFT) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: Was commenting on the wrong method. Let me try this one again.

Comment (jkoritzinsky) [reply src/libraries/Common/src/Interop/Unix/System.Native/Interop.EnumerateInterfaceAddresses.cs:56]: That only prevents the `Interop.IpHlpApi.GetIfEntry2` API from being changed to a blittable API.

I don't see anything that stops this API, `Interop.Sys.GetNetworkInterfaces` from being changed to a blittable signature.

**Review by jkoritzinsky — APPROVED** [review]

**Pull request merged**